ui/src/components/Sessions/index.js

```diff
@@ -1,14 +1,42 @@
-import React from "react";
+import React, { useState } from "react";
 import {Link} from "react-router-dom";
-
-import "./style.scss";
-
+import DeleteIcon from '@material-ui/icons/Delete'
 import {CSSTransition, TransitionGroup,} from 'react-transition-group';
+import { Observable } from "rxjs";
+import "./style.scss";
+import CircularProgress from '@material-ui/core/CircularProgress';
 
 const Sessions = (props) => {
     const {sessions = {}} = props;
     const list = Object.keys(sessions);
 
+    const [loading, setLoading] = useState([])
+
+    const deleteSession = (e, id) => {
+      e.preventDefault()
+      e.stopPropagation()
+      setLoading([...loading, id])
```

Comment:
please don't use state, pass the callback via rx-connect - you can see a great tutorial on that 
https://bsideup.gitbooks.io/rxconnect/content/

so the components became stateless - you can spot an example in the Session/index.js component